mlrun/utils/vault.py

```diff
@@ -0,0 +1,269 @@
+        )
+
+        response = self._vault_api_call("DELETE", url)
+        if not response:
+            raise ValueError(
+                f"Vault failed the API call to delete secrets. project={project}/user={user}"
+            )
+
+    def create_project_policy(self, project):
+        policy_name = "mlrun-project-{}".format(project)
+        # TODO - need to make sure name is escaped properly and invalid chars are stripped
+        url = "v1/sys/policies/acl/" + policy_name
+
+        policy_str = (
+            'path "secret/data/mlrun/projects/{0}" {{\n'
+            + '  capabilities = ["read", "list", "create", "delete", "update"]\n'
+            + "}}\n"
+            + 'path "secret/data/mlrun/projects/{0}/*" {{\n'
+            + '  capabilities = ["read", "list", "create", "delete", "update"]\n'
+            + "}}"
+        ).format(project)
+
+        data_object = {"policy": policy_str}
+
+        response = self._vault_api_call("PUT", url, data_object)
+        if not response:
+            raise ValueError(
+                "Vault failed the API call to create a policy. Response code: ({}) - {}".format(
+                    response.status_code, response.reason
+                )
+            )
+        return policy_name
+
+    def create_project_role(self, project, sa, policy, namespace="default-tenant"):
+        role_name = "mlrun-role-project-{}".format(project)
+        # TODO - need to make sure name is escaped properly and invalid chars are stripped
+        url = "v1/auth/kubernetes/role/" + role_name
+
+        role_object = {
+            "bound_service_account_names": sa,
+            "bound_service_account_namespaces": namespace,
+            "policies": [policy],
+            "token_ttl": 1800000,
+        }
+
+        response = self._vault_api_call("POST", url, role_object)
+        if not response:
+            raise ValueError(
+                "Vault failed the API call to create a secret. Response code: ({}) - {}".format(
+                    response.status_code, response.reason
+                )
+            )
+        return role_name
+
+
+def add_vault_project_secrets(project, items):
+    return VaultStore().add_vault_secrets(items, project=project)
+
+
+def add_vault_user_secrets(user, items):
+    return VaultStore().add_vault_secrets(items, user=user)
+
+
+def init_project_vault_configuration(project):
+    """Create needed configurations for this new project:
+    - Create a k8s service account with the name sa_vault_{proj name}
+    - Create a Vault policy with the name proj_{proj name}
+    - Create a Vault k8s auth role with the name role_proj_{proj name}
+    These constructs will enable any pod created as part of this project to access the project's secrets
+    in Vault, assuming that the secret which is part of the SA created is mounted to the pod.
+
+    :param project: Project name
+    """
+    logger.info(f"init_project_vault_configuration called, project name: {project}")
```

Comment:
* In logs try to be as readable as possible - "Initializing project vault configuration"
* Also don't format variables to the string, simply use kwargs